enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,346 @@
+  - The Backup Pod will be scheduled always to this master node where the backup data exists.
+  - Using a separate disk for backups (e.g. SSD) is possible, since local volume allows mounting it into a Pod.
+  - Having the deterministic Pod scheduling is vital, since the restoration needs the recent backup in place to work.
+* Cons
+  - If the master node where the `PV` is mounted became unhealthy/unavailable/unreachable, then the backups are no longer accessible, also taking new backups is no longer possible.
+
+#### Host Path volumes
+
+A `hostPath` volume mounts a path from the node's file system as a volume into a Pod.
+
+* Pros
+  - It supports all Openshift variants, including `SNO` and `BM`.
+* Cons
+  - `hostPath` could have security impact as it exposes the node's filesystem.
+  - No scheduling guarantees for the pod using `hostpath` as is the case with `localVolume`. The pod could be scheduled on a different node from where the hostPath volume exist.
+
+### Hybrid approach
+
+A hybrid solution is possible, in which dynamic provisioning is being used in cloud based Openshift, while utilising local storage for BM and SNO.
+Relying on `infrastructure` resource type, we can create the storage programmatically according to the underlying infrastructure and Openshift variant.
+
+### StatefulSet approach
+
+A `StatefulSet` could be deployed among all master nodes, where each backup pod has its own `PV`. This approach has the pros of spreading the backups among all master nodes.
+The complexity come from the fact that the backups are being triggered by a `CronJob` which spawn a `Job` to take the actual backup, by deploying a Pod.
+Since StatefulSet manages its own Pods, it is not possible to schedule backups by a `CronJob`. However, it is possible to generate event by the `CronJob` which is being watched by the StatefulSet. Then the backups are being taken and controlled mainly by it.
+
+### Parallel Jobs approach
+
+The `CronJob` based implementation schedules a `Job` upon each trigger to take the backup.
+An attempt has been investigated, which exploits [Parallel Jobs](https://kubernetes.io/docs/concepts/workloads/controllers/job/#parallel-jobs).
+Using NodeAffinity and PodAntiAffinity, while setting the Parallelism to the number of master nodes within an Openshift cluster, it is guaranteed that a Backup Job is being scheduled per each master node, while no more than one Backup Job is being scheduled to the same master node.
+Since the `config.openshift.io/v1alpha1 Backup` API expects only one `PVC`, only one Backup Job could mount the PVC and schedule the backup.
+A workaround has been tried, in which the `job.Spec` mounts a `hostPath` volume, so that it is being used on each master node independently.
+However, since the `Backup` job runs two containers (i.e. request-backup and prune) and the containers expect `PVC` as an argument, the approach has been declared as non-viable.
```

Comment:
I don't think this is accurate based on the latest testing and discussion is it?